source/d3d11/runtime_d3d11.cpp

```diff
@@ -128,7 +157,7 @@ bool reshade::d3d11::runtime_d3d11::on_init()
 	_backbuffer_format = swap_desc.BufferDesc.Format;
 
 	// Get back buffer texture
-	if (FAILED(_swapchain->GetBuffer(0, IID_PPV_ARGS(&_backbuffer))))
+	if (FAILED(swapchain->GetBuffer(0, IID_PPV_ARGS(&_backbuffer))))
```

Comment:
You may want to check out the code around https://github.com/crosire/reshade/blob/b505e1baf4cbf81ca257b8b0faf49102ffecd854/source/d3d12/runtime_d3d12.cpp#L213 on how this could be done without the code duplication in `on_init_vr`.

Reply:
I extracted the common code to an `init_resources` function and removed vr specific functions. As you suggested, you can now pass a texture to `on_present` instead, to be used as the backbuffer.